### PR TITLE
Mutable RedBlackTree improvements, tests & benchmarks

### PR DESCRIPTION
**Pull request by szeiger**
This is a byproduct of my work on the new OrderAwareHashMap before I
decided to fork this code to create a more customized implementation.

### CODE REVIEW

**Comment by Ichoran** [comment]
Do you have benchmark results to post?  Some of the changes are hard to gauge by eye.

**Comment by szeiger** [comment]
Here are some benchmark results:

Old:

```
[info] Benchmark                          (size)  Mode  Cnt        Score       Error  Units
[info] RedBlackTreeBenchmark.build            10  avgt   20      334.606 ±     2.717  ns/op
[info] RedBlackTreeBenchmark.build           100  avgt   20     3543.328 ±    38.076  ns/op
[info] RedBlackTreeBenchmark.build          1000  avgt   20    50187.963 ±   344.337  ns/op
[info] RedBlackTreeBenchmark.build         10000  avgt   20   667979.330 ±  7868.570  ns/op
[info] RedBlackTreeBenchmark.buildRandom      10  avgt   20      193.917 ±     2.331  ns/op
[info] RedBlackTreeBenchmark.buildRandom     100  avgt   20     2756.431 ±    26.815  ns/op
[info] RedBlackTreeBenchmark.buildRandom    1000  avgt   20    86567.983 ±   940.563  ns/op
[info] RedBlackTreeBenchmark.buildRandom   10000  avgt   20  1441076.833 ± 14552.327  ns/op
[info] RedBlackTreeBenchmark.copy             10  avgt   20      368.513 ±    12.807  ns/op
[info] RedBlackTreeBenchmark.copy            100  avgt   20     7010.045 ±   217.398  ns/op
[info] RedBlackTreeBenchmark.copy           1000  avgt   20    77487.917 ± 22794.373  ns/op
[info] RedBlackTreeBenchmark.copy          10000  avgt   20   719029.584 ±  7543.513  ns/op
[info] RedBlackTreeBenchmark.copyDrain        10  avgt   20      557.624 ±    41.170  ns/op
[info] RedBlackTreeBenchmark.copyDrain       100  avgt   20     9197.863 ±   155.789  ns/op
[info] RedBlackTreeBenchmark.copyDrain      1000  avgt   20   234537.682 ±  3931.420  ns/op
[info] RedBlackTreeBenchmark.copyDrain     10000  avgt   20  3638408.791 ± 48537.322  ns/op
[info] RedBlackTreeBenchmark.foreach          10  avgt   20       24.041 ±     0.211  ns/op
[info] RedBlackTreeBenchmark.foreach         100  avgt   20      345.396 ±     3.546  ns/op
[info] RedBlackTreeBenchmark.foreach        1000  avgt   20     3801.345 ±    35.156  ns/op
[info] RedBlackTreeBenchmark.foreach       10000  avgt   20    50964.446 ±   623.269  ns/op
```

New:

```
[info] Benchmark                          (size)  Mode  Cnt        Score       Error  Units
[info] RedBlackTreeBenchmark.build            10  avgt   20      136.804 ±     1.325  ns/op
[info] RedBlackTreeBenchmark.build           100  avgt   20     1122.636 ±     9.086  ns/op
[info] RedBlackTreeBenchmark.build          1000  avgt   20    11187.436 ±   325.154  ns/op
[info] RedBlackTreeBenchmark.build         10000  avgt   20   129842.399 ±  8645.900  ns/op
[info] RedBlackTreeBenchmark.buildRandom      10  avgt   20      245.728 ±     5.757  ns/op
[info] RedBlackTreeBenchmark.buildRandom     100  avgt   20     2732.306 ±    26.747  ns/op
[info] RedBlackTreeBenchmark.buildRandom    1000  avgt   20    79369.396 ±  1717.383  ns/op
[info] RedBlackTreeBenchmark.buildRandom   10000  avgt   20  1425532.495 ± 12188.363  ns/op
[info] RedBlackTreeBenchmark.copy             10  avgt   20      107.725 ±     1.194  ns/op
[info] RedBlackTreeBenchmark.copy            100  avgt   20      932.578 ±    25.179  ns/op
[info] RedBlackTreeBenchmark.copy           1000  avgt   20    12055.298 ±   689.872  ns/op
[info] RedBlackTreeBenchmark.copy          10000  avgt   20   112780.842 ±  8702.459  ns/op
[info] RedBlackTreeBenchmark.copyDrain        10  avgt   20      409.419 ±     6.962  ns/op
[info] RedBlackTreeBenchmark.copyDrain       100  avgt   20     6623.767 ±   219.140  ns/op
[info] RedBlackTreeBenchmark.copyDrain      1000  avgt   20   183040.519 ±  1313.614  ns/op
[info] RedBlackTreeBenchmark.copyDrain     10000  avgt   20  3038500.443 ± 58327.606  ns/op
[info] RedBlackTreeBenchmark.foreach          10  avgt   20       47.835 ±     0.885  ns/op
[info] RedBlackTreeBenchmark.foreach         100  avgt   20      259.290 ±     2.194  ns/op
[info] RedBlackTreeBenchmark.foreach        1000  avgt   20     3341.914 ±    55.069  ns/op
[info] RedBlackTreeBenchmark.foreach       10000  avgt   20    45493.514 ±  1445.428  ns/op
```

**Comment by szeiger** [comment]
Does anyone else want to review this? It should be rather uncontroversial and Rex still has plenty of other PRs to look after.

File: src/library/scala/collection/mutable/RedBlackTree.scala [hunk 1]
```diff
@@ -17,6 +17,8 @@ import scala.annotation.tailrec
 import collection.{AbstractIterator, Iterator}
 import java.lang.String
 
+import scala.annotation.meta.getter
```
Comment (SethTisue) [inline src/library/scala/collection/mutable/RedBlackTree.scala]: unused import

**Review by SethTisue — CHANGES_REQUESTED** [review]
look, a nit!

otherwise LGTM

nice property-based tests

scala-collection-laws will also get a whack at this, post-merge.

**Review by SethTisue — APPROVED** [review]

**Comment by SethTisue** [comment]
I hit "restart" on Travis

**Pull request merged**